mountpoint-s3/src/fuse.rs

```diff
@@ -182,7 +187,7 @@ where
 
         let replier = ReplyDirectoryPlus { inner: &mut reply };
 
-        match block_on(self.fs.readdir(parent, fh, offset, replier).in_current_span()) {
+        match block_on(self.fs.readdir(parent, fh, offset, true, replier).in_current_span()) {
```

Comment:
I agree, the flag is not great. But I cannot immediately see how else to avoid too much duplication between `readdir` and `readdirplus`, besides pushing the flag down to a `readdir_common` they both call. Suggestions welcome... or I'll give it another go tomorrow.

Reply:
I think pushing the flag down into a common readdir in `fuse.rs` is fine. The fact that we were reusing `readdir` here is really just laziness -- in reality, `fs.rs` needs to know whether it's readdir or readdirplus, so it should just have both.